docs/source/guides/upload.mdx

```diff
@@ -264,9 +258,62 @@ For more details about the [`CommitScheduler`], here is what you need to know:
     `scheduler.lock` lock to ensure thread-safety. The lock is blocked only when the scheduler scans the folder for
     changes, not when it uploads data. You can safely assume that it will not affect the user experience on your Space.
 
-### create_commit
+### Space persistence demo
+
+Persisting data from a Space to a Dataset on the Hub is the main use case for [`CommitScheduler`]. Depending on the use
+case, you might want to structure your data differently. The structure has to be robust to concurrent users and
+restarts which often implies generating uuids. Beside robustness, you would ideally like to upload data in a format that
+is readable by the `datasets` library for later reuse. We compiled in this [interactive demo Space](https://huggingface.co/spaces/Wauplin/space_to_dataset_saver)
+implementations to tackle major use cases. Depending on your specific needs, you might need to adapt them
 
-If you want to work at a commit-level, use the [`create_commit`] function directly. There are two types of operations supported by [`create_commit`]:
+### Custom uploads
+
+As said above, [`CommitScheduler`] assumes that your data is append-only and should be uploading "as is". However, you
+might want to customize the way data is uploaded. You can do that by creating a class inheriting from [`CommitScheduler`]
+and overwrite the `push_to_hub` method. You can overwrite this method the way you want. You are guaranteed that it will
```

Comment:
```suggestion
might want to customize the way data is uploaded. You can do that by creating a class inheriting from [`CommitScheduler`]
and overwrite the `push_to_hub` method (feel free to overwrite it any way you want). You are guaranteed it will
```